File: ocaml/01-10/05.ml
(* Reverse a list. (easy) *)

(* rev : 'a list -> 'a list *)
let rec rev = function
  | [] -> []
  | x::xs -> (rev xs) @ [x]

let _ = assert ((rev ["a" ; "b" ; "c"]) = ["c"; "b"; "a"])

File: prolog/01-10/05.pl
/*
P05 (*) Reverse a list.
*/
revacc([], Ys, Ys).
revacc([H|T], Ys, Acc) :- revacc(T, Ys, [H|Acc]).

rev([], []).
rev(Xs, Ys) :- revacc(Xs, Ys, []). 
